overlord/snapstate/backend/backend.go

```diff
@@ -20,5 +20,26 @@
 // Package backend implements the low-level primitives to manage the snaps and their installation on disk.
 package backend
 
+import (
+	"github.com/snapcore/snapd/snap"
+)
+
 // Backend exposes all the low-level primitives to manage snaps and their installation on disk.
 type Backend struct{}
+
+// OpenSnapFile opens a snap blob returning both a snap.Info completed
+// with sideInfo (if not nil) and a corresponding snap.Container.
+// Assumes the file was verified beforehand or the user asked for devmode.
+func OpenSnapFile(snapPath string, sideInfo *snap.SideInfo) (*snap.Info, snap.Container, error) {
```

Comment:
:heart: I will use this in my feature/firstboot branch - exactly what I need there.